crates/wasmtime/src/compile/code_builder.rs

```diff
@@ -139,6 +149,53 @@ impl<'a> CodeBuilder<'a> {
         Ok((&wasm[..]).into())
     }
 
+    /// Explicitly specify DWARF `.dwp` path.
+    pub fn dwarf_package_file(&mut self, file: &Path) -> Result<&mut Self> {
+        if self.dwarf_package.is_some() {
+            bail!("cannot call `dwarf_package` or `dwarf_package_file` twice");
+        }
+
+        if !file.exists() {
+            bail!("DWARF package file does not exist");
+        }
+
+        let dwarf_package = std::fs::read(file)
+            .with_context(|| format!("failed to read dwarf input file: {}", file.display()))?;
+        self.dwarf_package = Some(dwarf_package.into());
+
+        Ok(self)
+    }
+
+    fn dwarf_package_from_wasm_path(&mut self) -> Result<&mut Self> {
+        let dwarf_package_path_buf = self.wasm_path.as_ref().unwrap().with_extension("dwp");
+        if dwarf_package_path_buf.exists() {
+            self.dwarf_package_path = Some(Cow::Owned(dwarf_package_path_buf.as_path().to_owned()));
+            let path = dwarf_package_path_buf.as_path();
+            return self.dwarf_package_file(path);
+        }
+
+        Ok(self)
+    }
+
+    /// Gets the DWARF package.
+    pub fn dwarf_package_binary(&self) -> &Option<Cow<'_, [u8]>> {
```

Comment:
Mind making this not-`pub`? (I think it's only intended for internal bits)

As a minor nit here this might most appropriately return `Option<&[u8]>` as well, but that's just a minor thing.

Reply:
Its used in https://github.com/bytecodealliance/wasmtime/pull/8055/files#diff-92db9b7b9b675f97d5c8d9d94e0680d63aab09fb1454f5441cb61ddbff8d3d81R17

Have changed the return type, thanks.